### DISCUSSION TITLE
successfully intall fortls in linux but don't work in VS code using ssh

### DISCUSSION THREAD

**Discussion by GYC-lab** [fortran-lang/fortls#191] [Q&A] [answered]
**Description**
I use ssh in my PC's VS code(which is installed in Win11) to connect the supercomputer whose system is linux. When I use `fortls --version`, it returns the correct result `2.13.0`, which means fortls is installed successfully. But when I view my code in my PC's VS code, it still reports no fortls.

So how can I make the fortls actually work and enjoy it on my PC's VS code? Or **how to make my PC's VS code installed in Win11 loacte fortls installed in the linux**? 

Best Regards!

**Version of Modern Fortran**
v3.2.0

**Version of Visual Studio Code**
v1.70.2

**Platform and Architecture**
Linux

**Comment by gnikit** [comment, marked as answer]
With VS Code's SSH extensions what you end up doing is install a copy of vscode and all the extensions you are using in your remote machine and your local machine simply connects to that. So to get `fortls` on your **remote machine** you actually need to install it **in your remote machine** with `pip`, see the Docs: https://gnikit.github.io/fortls/quickstart.html#download. Assuming that your windows machine has the pip install location to its PATH you should be good to go, else you need to set the `"fortran.fortls.path": "fortls"` setting to the absolute path to `fortls`.

**Reply by GYC-lab** [reply]
Thanks for your reply! 
I have tried to set the "fortran.fortls.path": "fortls" setting to the absolute path to fortls, and it seemed that, my PC's VS code can locate fortls successfully. However, there is an error about launching fortls, like this:
![image](https://user-images.githubusercontent.com/89847334/190662225-7f0484ef-3f8c-4d83-a0c9-8703fb041490.png)

**Reply by GYC-lab** [reply]
FYI, the version of fortls is 2.13.0.

**Reply by gnikit** [reply]
That message says that you have successfully launch `fortls`, but some of the options you input are wrong. These are all the valid configuration options https://gnikit.github.io/fortls/options.html what are your `fortls` options in your remote machine?

**Reply by GYC-lab** [reply]
OK. FYI, I haven't set any configuration of  `fortls` yet. I will try it refering to https://gnikit.github.io/fortls/options.html. And thanks again*^____^*

**Reply by gnikit** [reply]
No worries, I would do a sanity check on your remote machine to test that `fortls` is correctly picked up.

for Linux/OS X

```sh
which fortls
```

or Windows

```cmd
where fortls
```

and then if `fortls` is found I would go to a Fortran file and try something like

```sh
fortls --debug_filepath your_file.f90 --debug_rootpath . --debug_parser
```

Is this yields an output then you are good.
Another place to check is the Log Channel in VS Code, check `Modern Fortran` and see if you see anything suspicious in the output.
